Sources/SPTDataLoader.m

```diff
@@ -201,13 +241,19 @@ - (void)cancelledRequest:(SPTDataLoaderRequest *)request
     if (![self isRequestExpected:request]) {
         return;
     }
-
     if ([self.delegate respondsToSelector:@selector(dataLoader:didCancelRequest:)]) {
-        [self executeDelegateBlock: ^{
+        [self executeDelegateBlock:^{
             [self.delegate dataLoader:self didCancelRequest:request];
         }];
+    } else {
+        [self executeDelegateBlock:^{
+            void (^handler)(SPTDataLoaderResult *) = [self.requestHandlers objectForKey:request];
+            handler([SPTDataLoaderResult fromRequest:request kind:SPTDataLoaderResultCancelled]);
```

Comment:
Always check for `nil` before invoking a block. If `handler == nil`, this will crash.